version: '3'
services:
  server:
    shm_size: 3GB
#    deploy:
#      resources:
#        limits:
#          cpus: '3'
#        reservations:
#          cpus: '3'
#      restart_policy:
#        condition: on-failure
    volumes:
      - ./config.py:/fed-flow/app/config/config.py
    image: negaranabestani/fed-flow:rabbitmq
    hostname: server
    ports:
      - "51008:51008"
    working_dir: /fed-flow/app/fl_training/runner
    entrypoint: bash -c "python3 /fed-flow/app/fl_training/runner/fed_server_run.py -m alexnet"
  broker:
    attach: false
    volumes:
      - ./rabbitmq.conf:/etc/rabbitmq/rabbitmq.conf
    image: rabbitmq:3.13-management
    ports:
      - "5672:5672"
      - "15672:15672"
    environment:
      - RABBITMQ_DEFAULT_USER=user
      - RABBITMQ_DEFAULT_PASS=password
    hostname: broker

  client1:
    shm_size: 3GB
#    deploy:
#      resources:
#        limits:
#          cpus: '0.5'
#        reservations:
#          cpus: '0.5'
#      restart_policy:
#        condition: on-failure
    volumes:
      - ./config.py:/fed-flow/app/config/config.py
    image: negaranabestani/fed-flow:rabbitmq
    hostname: client1
    working_dir: /fed-flow/app/fl_training/runner
    entrypoint: bash -c "python3 /fed-flow/app/fl_training/runner/fed_client_run.py -i 0 -m alexnet & python3 /fed-flow/energy/service.py & wait -n"

